action.yml

```diff
@@ -234,6 +239,21 @@ runs:
           fi
         fi
 
+    - if: ${{ inputs.cache == 'true' }}
+      id: gen-cache-key
+      shell: python {0}
+      run: |
+        import hashlib, os, pathlib
+        bytes = pathlib.Path(os.path.join(os.environ["GITHUB_ACTION_PATH"], 'requirements.txt')).read_bytes()
+        print('::set-output hash=%s' % (hashlib.sha1(bytes).hexdigest()))
```

Comment:
This is not how outputs are set any more. This approach was deprecated for security reasons. Instead, we write to the path indicated by `os.environ["GITHUB_OUTPUT"]`:
```python
        if "GITHUB_OUTPUT" in os.environ:
            with open(os.environ["GITHUB_OUTPUT"], "a", encoding="utf-8") as env_file:
                env_file.write(f"hash={hashlib.sha1(bytes).hexdigest()}\n")
```